XSConsoleUtils.py

```diff
@@ -190,28 +190,29 @@ def DateTimeToSecs(cls, inDateTime):
         return retVal
 
 class IPUtils:
+    @classmethod
+    def ValidateIPFamily(cls, text, family):
+        try:
+            socket.inet_pton(family, text)
+            return True
+        except socket.error:
+            return False
+
+    @classmethod
+    def ValidateIPv4(cls, text):
+        return cls.ValidateIPFamily(text, socket.AF_INET)
+
+    @classmethod
+    def ValidateIPv6(cls, text):
+        return cls.ValidateIPFamily(text, socket.AF_INET6)
+
     @classmethod
     def ValidateIP(cls, text):
-        rc = re.match("^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)$", text)
-        if not rc: return False
-        ints = list(map(int, rc.groups()))
-        largest = 0
-        for i in ints:
-            if i > 255: return False
-            largest = max(largest, i)
-        if largest == 0: return False
-        return True
+        return cls.ValidateIPv4(text) or cls.ValidateIPv6(text)
 
     @classmethod
     def ValidateNetmask(cls, text):
-        rc = re.match("^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)$", text)
-        if not rc:
-            return False
-        ints = list(map(int, rc.groups()))
-        for i in ints:
-            if i > 255:
-                return False
-        return True
+        return cls.ValidateIPv4(text) or (int(text) > 4 and int(text) < 128)
```

Comment:
The previous code accepted `0.0.0.0` in addition to a valid host IP, this check seems much wider.

In the long term I'd rather adopt [ipaddress](https://docs.python.org/3/library/ipaddress.html) now that we aren't hold back by python 2